segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleProcessor.java

```diff
@@ -178,19 +179,37 @@ private void triggerScaleDown(String streamSegmentName, boolean silent) {
     }
 
     private CompletableFuture<Void> writeRequest(ScaleEvent event) {
+        if (writer.get() == null) {
+            log.warn("Event writer not set up yet. Ignoring the request");
```

Comment:
it shouldnt happen, but in the previous PR you gave a comment about null check. i will make this assert.